### PR TITLE
Dummy model + frontend + Telegram Bot

### PR DESCRIPTION
**Pull request by berlinikki**
I've pre-merged my three branches for simplicity, so all new code additions to master are now in dummy-model. Pls review

### CODE REVIEW

File: api/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.env
+Vincent Van Bot-67b2b35a7d0c.json
```
Comment (seve-26) [inline api/.gitignore:2]: Even if git ignored, the name of the JSON file is still visible, which includes some sort of key. I would consider placing this in a separate folder

File: api/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM python:3.8-slim
+COPY requirements.txt ./requirements.txt
+RUN pip install -r requirements.txt
+COPY . ./
+
+ENV GOOGLE_APPLICATION_CREDENTIALS='Vincent Van Bot-67b2b35a7d0c.json'
+
+CMD uvicorn api:app --host 0.0.0.0 --port $PORT
```
Comment (seve-26) [inline api/Dockerfile:8]: To adhere to standard style, we might want to add a blank new line at the end of each file

File: api/api.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    
+    urls = dummy_model(file.file)
+    return urls
+
+
+def dummy_model(file):
+    
+    GOOGLE_SEARCH_API_LINK='https://customsearch.googleapis.com/customsearch/v1'
+    api_key = os.getenv('API_KEY')
+    cx_key = os.getenv('CX_KEY')
```
Comment (seve-26) [inline api/api.py:37]: Are these keys something we can directly add to repo secrets? We'll need to test how to make it work on different laptops

File: api/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+twine
+wheel>=0.29
+
+streamlit
+fastapi
+google-cloud-vision
+python-multipart
+python-dotenv
+requests
+uvicorn
```
Comment (seve-26) [inline api/requirements.txt:12]: We will need to merge this file with the main requirements.txt . @rtnikki do you think it is better to wait until your part is done?

File: frontend/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+.env
```
Comment (seve-26) [inline frontend/.gitignore:1]: What does this `.env` file include? Is it the file containing your keys?

File: frontend/app.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import os
+from os.path import join, dirname
+from dotenv import load_dotenv
+#from PIL import Image
+#import copy
+
+env_path = join(dirname(__file__),'.env')
+load_dotenv(dotenv_path=env_path)
+
+api_url = os.getenv('API_URL')
```
Comment (seve-26) [inline frontend/app.py:13]: Should this `API_URL` go to a params.py file to be consistent across the package?

File: tbot/bot.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    bot.reply_to(message, "Hello! Just send me a picture")
+    
+@bot.message_handler(content_types=UNSUPPORTED_TYPES)
+def default_reply(message):
+    bot.send_message(message.chat.id, "Unfortunately I cannot work with this data type 😢")
+    bot.send_message(message.chat.id, 'Send me a picture 😅')
+ 
+@bot.message_handler(content_types=['text'])
+def reply_text(message):
+    bot.reply_to(message, "I'm not very talkative. Better send me a picture 😅")
```
Comment (seve-26) [inline tbot/bot.py:33]: Once we work on the painting recommender system, this will actually be a valid input ;)

File: tbot/bot.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            while (len(pic_received) > 10000000):
+                img = Image.open(io.BytesIO(pic_received))
+                img = img.resize((img.size[0] // 2, img.size[1] // 2), Image.ANTIALIAS)
+            
+                byteIO = io.BytesIO()
+                img.save(byteIO, format='PNG')
+                pic_received = byteIO.getvalue()
+        
+            bot.send_photo(message.chat.id, pic_received)
+            bot.send_message(message.chat.id, "URL: " + url, disable_web_page_preview=True)
```
Comment (seve-26) [inline tbot/bot.py:64]: How is readability on the Telegram app if we display URLs as well? Once ready, we might want to replace this with other info such as title and author

**Review by seve-26 — COMMENTED** [review]
Amazing work! Especially the tbot part I found it super well written. One could use a bit more comments in the other sections, but it is still very clear. Thank you!

Just have a look at the comments, and once done we can merge 👍

File: api/Dockerfile (hunk 2 above)
Comment (berlinikki) [reply api/Dockerfile:8]: Ok, will pay attention to it :)

File: api/api.py (hunk 3 above)
Comment (berlinikki) [reply api/api.py:37]: These are necessary only for the dummy model to work. I can send .env files (work with them was described by Ben during the lecture on Monday)

File: api/requirements.txt (hunk 4 above)
Comment (berlinikki) [reply api/requirements.txt:12]: We actually don't need to merge. API server should be fully independent of the main model and should just use pickle file to 'connect' to the model (same as we did for the taxi thing last week)

File: frontend/.gitignore (hunk 5 above)
Comment (berlinikki) [reply frontend/.gitignore:1]: I've hidden the link to our API server there. Will send over the files: there are three .env files: one for frontend, one for api server, one for telegram bot

File: frontend/app.py (hunk 6 above)
Comment (berlinikki) [reply frontend/app.py:13]: Good idea about this being accessible in one place - but I would still prefer to have in .env file. We can have one .env for the whole project - I will try to refactor the code in the next version to make it work

File: tbot/bot.py (hunk 7 above)
Comment (berlinikki) [reply tbot/bot.py:33]: Yes :)

File: tbot/bot.py (hunk 8 above)
Comment (berlinikki) [reply tbot/bot.py:64]: Yeah, to be discussed of course :) I just decided to send urls to have some text fields, not just pics

File: api/requirements.txt (hunk 4 above)
Comment (seve-26) [reply api/requirements.txt:12]: Ok perfect!

File: api/api.py (hunk 3 above)
Comment (seve-26) [reply api/api.py:37]: 👍

File: frontend/app.py (hunk 6 above)
Comment (seve-26) [reply frontend/app.py:13]: Ok! Let's keep this in mind, we can then move everything to a `.env`

**Pull request merged**